lib/agent.js

```diff
@@ -102,6 +95,18 @@ class Agent extends DispatcherBase {
       this[kClients].set(key, dispatcher)
     }
 
+    if (this[kMaxRedirections] > 0) {
+      return dispatcher.dispatch(
+        opts,
+        new RedirectHandler(
+          dispatcher.dispatch.bind(dispatcher),
+          this[kMaxRedirections],
+          opts,
+          handler
+        )
+      )
+    }
```

Comment:
I would prefer to move this one level higher to the api methods, i.e. totally exclude from normal dispatch

Reply:
So basically just expose it but not add it directly; then several tests will be affected, I’ll go over them during the weekend then 👍 

I’ll split this into two PRs, one to remove all sign of interceptors, and a second one to expose Redirect handler and others in a different way.

Reply:
You don't actually need to break so many tests if you just move it into the api methods (stream, pipeline and request).

Reply:
Ah, ok, I didn't get that first; so instead of keeping it at a low level, you suggest adding it to the APIs instead; if I understood correctly, sure 👍